docs/guides/self_hosting.md

```diff
@@ -0,0 +1,117 @@
+
+Ensure that Helm V3 is installed [instructions](https://helm.sh/docs/intro/install/) and can connect to the EKS cluster. Users should be able to install the chart with `helm install llm-engine llm-engine -f llm-engine/values_sample.yaml -n <DESIRED_NAMESPACE>`.
+Below are the configurations to specify in the `values_sample.yaml` file. 
+
+| Parameter | Description | Required |
+| --- | --- | --- |
+| tag | The LLM Engine docker image tag | Yes |
+| context | A user-specified deployment tag | No |
+| image.gatewayRepository | The docker repository to pull the LLM Engine gateway image from | Yes |
+| image.builderRepository | The docker repository to pull the LLM Engine endpoint builder image from | Yes |
+| image.cacherRepository | The docker repository to pull the LLM Engine cacher image from | Yes |
+| image.forwarderRepository | The docker repository to pull the LLM Engine forwarder image from | Yes |
+| image.pullPolicy | The docker image pull policy | No |
+| secrets.kubernetesDatabaseSecretName | The name of the secret that contains the database credentials | Yes |
+| serviceAccount.annotations.eks.amazonaws.com/role-arn | The ARN of the IAM role that the service account will assume | Yes |
+| service.type | The service configuration for the main LLM Engine server | No |
+| service.port | The service configuration for the main LLM Engine server | No |
+| replicaCount | The amount of replica pods for each deployment | No |
+| autoscaling | The autoscaling configuration for LLM Engine server deployments | No |
+| resources.requests.cpu | The k8s resources for LLM Engine server deployments | No |
+| nodeSelector | The node selector for LLM Engine server deployments | No |
+| tolerations | The tolerations for LLM Engine server deployments | No |
+| affinity | The affinity for LLM Engine server deployments | No |
+| datadog_trace_enabled | Whether to enable datadog tracing, datadog must be installed in the cluster | No |
+| aws.configMap.name | The AWS configurations (by configMap) for LLM Engine server deployments | No |
+| aws.configMap.create | The AWS configurations (by configMap) for LLM Engine server deployments | No |
+| aws.profileName | The AWS configurations (by configMap) for LLM Engine server deployments | No |
+| serviceTemplate.securityContext.capabilities.drop | Additional flags for model endpoints | No |
+| serviceTemplate.mountInfraConfig | Additional flags for model endpoints | No |
+| config.values.infra.k8s_cluster_name | The name of the k8s cluster | Yes |
+| config.values.infra.dns_host_domain | The domain name of the k8s cluster | Yes |
+| config.values.infra.default_region | The default AWS region for various resources | Yes |
+| config.values.infra.ml_account_id | The AWS account ID for various resources | Yes |
+| config.values.infra.docker_repo_prefix | The prefix for AWS ECR repositories | Yes |
```

Comment:
Is this ECR-specific? If so, we might want to prefix with `ecr_` or `aws_ecr_`.

Reply:
It is, but we will need to do some extra find + replace in the code. I think we can use this as a follow up.